src/main/java/in/projecteka/consentmanager/common/cache/ICacheAdapter.java

```diff
@@ -0,0 +1,11 @@
+package in.projecteka.consentmanager.common.cache;
+
+public interface ICacheAdapter<K,V> {
```

Comment:
```suggestion
public interface ICache<K,V> {
```

Reply:
`ICache` feels a bit too generic, whereas an Adaptor explicitly specifies its purpose.

Reply:
Understood, but when we define an Abstraction, we should not think about the implementation right?

Reply:
In this context, abstaction is itself an adaptor :-)

Reply:
Okay boss, if you want keep Adapter keep it. at least don't keep 'I' as a Prefix, that's not java convention right.

Reply:
Done.